### PR TITLE
Install and run `cargo watch` if user agrees

### PR DESCRIPTION
**Pull request by Xanewok**
This isn't a glorious patch but hopefully is useful :+1: This introduces a default background `cargo watch` task and (separately from that) asks the user on every startup if they want to run `cargo watch` (installs it if it's not available).

r? @matklad does it fit the what you've been thinking about?

### CODE REVIEW

File: editors/code/src/extension.ts [hunk 1]
```diff
@@ -129,3 +135,43 @@ export function deactivate(): Thenable<void> {
     }
     return Server.client.stop();
 }
+
+async function askToCargoWatch() {
```
Comment (Xanewok) [inline editors/code/src/extension.ts]: Should I move it to runnable.ts? It doesn't seem to *exactly* fit there but maybe it makes more sense since it's task-related and whatnot.

File: editors/code/src/extension.ts [hunk 2]
```diff
@@ -89,11 +92,11 @@ export function activate(context: vscode.ExtensionContext) {
     const allNotifications: Iterable<
         [string, lc.GenericNotificationHandler]
     > = [
-        [
-            'rust-analyzer/publishDecorations',
-            notifications.publishDecorations.handle
-        ]
-    ];
+            [
```
Comment (Xanewok) [inline editors/code/src/extension.ts]: Drive-by formatting applied when I formatted the entire document. Is that okay or should I revert it?

Comment (matklad) [reply editors/code/src/extension.ts]: I think the `.ts` formatting is nailed down by prettier in the CI

File: editors/code/src/extension.ts (hunk 1 above)
Comment (matklad) [reply editors/code/src/extension.ts]: Yeah, I think `runnable.ts` is maybe a good place. 

Overall, I am not too happy with the current organization of code of the extension: there are way to many files; However, it is tolerable, so I am constantly postponing reorganizing it :)

File: editors/code/src/extension.ts [hunk 3]
```diff
@@ -129,3 +135,43 @@ export function deactivate(): Thenable<void> {
     }
     return Server.client.stop();
 }
+
+async function askToCargoWatch() {
+    const watch = await vscode.window.showInformationMessage(
```
Comment (matklad) [inline editors/code/src/extension.ts]: I think we should have a config option here to suppress the notification.

**Comment by matklad** [comment]
Cool!

The only real comment I have is that all notifications/questions should have an option to disable them.

File: editors/code/src/extension.ts (hunk 3 above)
Comment (Xanewok) [reply editors/code/src/extension.ts]: So should this be entirely config-controlled? Either you have it enabled there and you are not prompted or you disable it completely or:
- enabled, so we ask you everytime
- disabled, so we never ask you?

File: editors/code/src/extension.ts (hunk 2 above)
Comment (Xanewok) [reply editors/code/src/extension.ts]: Did `yarn prettier --write`, this should fix any remaining formatting issues

File: editors/code/src/extension.ts (hunk 1 above)
Comment (Xanewok) [reply editors/code/src/extension.ts]: Right, makes sense :stuck_out_tongue: Moved this to `runnable.ts` then and renamed slightly

File: editors/code/src/extension.ts (hunk 3 above)
Comment (Xanewok) [reply editors/code/src/extension.ts]: Implemented the "when you explicitly turn off a config option we won't bother you anymore" option in https://github.com/rust-analyzer/rust-analyzer/pull/995/commits/7c2595c26820917fa9ad1b1c36f01fc6ac979287

**Comment by Xanewok** [comment]
Done, pushed the changes :+1:

**Comment by matklad** [comment]
bors r+

Thanks, will try this out tomorrow!
